### DISCUSSION TITLE
how to understand from where a function call is made?

### DISCUSSION THREAD

**Discussion by code2graph** [tree-sitter/tree-sitter#2520] [Q&A] [answered]
I am using TreeSitter to parse python codes. I need to determine from which file a function is called.

For example, I need to understand `check_files_in_directory` is invoked from `GPT4Readability.utils`. I already captured all the function calls.

But now I have to find out from which file `check_files_in_directory` is called. I am struggling to understand what would the logic to do it. Can anyone please suggest? 

```py
import os
from getpass import getpass
from GPT4Readability.utils import *
import importlib.resources as pkg_resources  


def generate_readme(root_dir, output_name, model):
    """Generates a README.md file based on the python files in the provided directory

    Args:
        root_dir (str): The root directory of the python package to parse and generate a readme for
    """

    # prompt_folder_name = os.path.join(os.path.dirname(__file__), "prompts")
    # prompt_path = os.path.join(prompt_folder_name, "readme_prompt.txt")

    with pkg_resources.open_text('GPT4Readability.prompts','readme_prompt.txt') as f:         
	    inb_msg = f.read()

    # with open(prompt_path) as f:
    #     lines = f.readlines()
    # inb_msg = "".join(lines)

    file_check_result = check_files_in_directory(root_dir)
```

**Comment by ahlinc** [comment, marked as answer]
You need to recursively parse all files that represent imports to find out where the function was defined.
Such tasks is out of scope for the Tree-sitter project, try to look on [tree-sitter-graph](https://github.com/tree-sitter/tree-sitter-graph), [stack-graphs](https://github.com/github/stack-graphs) that are based on Tree-sitter and look for similar projects.
